_ml-commons-plugin/model-serving-framework.md

```diff
@@ -0,0 +1,249 @@
+---
+layout: default
+title: Model serving framework 
+has_children: false
+nav_order: 110
+---
+
+# Model serving framework
+
+ML commons supports the ability to serve custom models and use those models to make inferences. For those who wish to use train their models using a deep-learning framework, such as Pytorch, TensorFlow, or ONNX, they can upload such models to OpenSearch in order to take advantage of OpenSearch resources, like GPU acceleration.
+
+This page outlines the steps required to upload a custom model with the ML Commons plugin.
+
+## Prerequisites 
+
+To upload a custom model to OpenSearch, you'll need to prepare outside of your OpenSearch cluster. You can use a current model or train a new model depending on your needs.
+
+### Model format
+
+Although most ML models are written in Python, you'll need to export the model into a portable format. As of 2.4, OpenSearch only supports [torchscript](https://pytorch.org/docs/stable/jit.html).
+
+### Model size
+
+Most deep-learning models are over 100 MBs, making it difficult to fit the model into a single document. OpenSearch splits in portable model file into smaller chunks to store in an OpenSearch index. When allocating ML or Data nodes to your OpenSearch cluster, please be aware of the size of your model to prevent any downtime when making inferences. Furthermore, asynchronous models load faster than synchronous.
+
+## Upload model to OpenSearch
+
+Use the URL upload operation for models that already exist on another server, such as GitHub or S3. 
+
+```
+POST /_plugins/_ml/models/_upload
+```
+
+The URL upload method requires the following request fields:
+
+Field | Data Type | Description
+:---  | :--- | :--- 
+`name`| string | The name of the model. |
+`version` | integer | The version number of the model. Version number should follow standard semantic format, major.minor.patch |
```

Comment:
What's the purpose of minor and patch versions for the model? Can you reference another system that uses this scheme? Typically, models only have a single version like how resource versioning typically works. It doesn't require a granular versioning system like software.

Reply:
@dylan-tong-aws: Good question! @ylwu-amzn, would you have any insight on this one?

Reply:
Hi, @Naarcha-AWS , the version should be string type, and we don't enforce the version format to follow `major.minor.patch`, user can choose their own version format like "v1", "v1.0" or simply "1", "1.0"

Reply:
```suggestion
`version` | string | The version number of the model. Since OpenSearch does not enforce a specific version schema for models, you can choose any number or format that makes sense for your models. |
```